functionalDB/chapter.md

```diff
@@ -520,20 +522,21 @@ We use two helper functions to perform the update. *update-attr-modification-tim
 
 ### Transactions
 
-Each of the operations in our low-level API act on a single entity. However, nearly all databases have a mechanism for allowing users to perform multiple operations as a single _transaction_. (TODO: Reference other chapters on transactional semantics here.) This means: 
+Each of the operations in our low-level API acts on a single entity. However, nearly all databases have a mechanism for allowing users to perform multiple operations as a single _transaction_. <!--TODO/FIXME: Reference other chapters on transactional semantics here.--> This means: 
+
+* The batch of operations is viewed as a single atomic operation, so all of the operations either succeed together or fail together.
+* The database is in a valid state before and after the transaction.
+* The batch update appears to be _isolated_; other queries should never see a database state in which only some of the operations have been applied.
 
-* The batch of operations is viewed as a single atomic operation, meaning that either all of the operations succeed together or fail together
-* The database is in a valid state before, and after the transaction
-* The batch update appears to be _isolated_; other queries should never see a database state in which only some of the operations have been applied
+We can fulfill these requirements through an interface that consumes a database and a set of operations to be performed, and produces a database whose state reflects the given changes. All of the changes submitted in the batch should be applied through the addition of a _single_ layer. However, we have a problem: All of the functions we wrote in our low-level API add a new layer to the database. If we were to perform a batch with _n_ operations, we would thus see _n_ new layers added, when what we would really like is to have exactly one new layer.   
 
-We can fulfill these requirements through an interface that consumes a database and a set of operations to be performed, and which produces a database whose state reflects the given changes. All of the changes submitted in the batch should be applied through the addition of a _single_ layer. However, we have one small problem: All of the functions we wrote in our low-level API add a new layer to the database. If we were to perform a batch with _n_ operations, we would thus see _n_ new layers added, when what we would really like is to have exactly 1 new layer.   
+The key here is that the layer we want is the _top_ layer that would be produced by performing those updates in sequence. Therefore, the solution is to execute the user’s operations one after another, each creating a new layer. When the last layer is created, we take only that top layer and place it on the initial database (leaving all the intermediate layers to pine for the fjords). Only after we've done all this will we update the database's timestamp.

```

Comment:
I'm so sorry, but I think we have to take out the "pine for the fjords" reference here. It will confuse anyone who doesn't get it, and for those who do it doesn't really make sense -- nothing here is dead or apparently dead, only left behind. If you can come up with a clever nerdy "left behind" reference I'll consider it. :)